predicators/explorers/active_sampler_explorer.py

```diff
@@ -213,7 +213,11 @@ def generate_goals() -> Iterator[Set[GroundAtom]]:
                 for goal in generate_goals():
                     task = Task(state, goal)
                     logging.info(f"[Explorer] Replanning to {task.goal}")
-
+                    # If the goal is empty, then we can just recursively
+                    # call the policy, since we don't need to execute
+                    # anything.
+                    if len(goal) == 0:
+                        return _option_policy(state)  # pragma: no cover
```

Comment:
i'm a little worried about adding this here if it's not covered. for example, are we certain that infinite recursion is impossible?